src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -9,32 +9,66 @@ import type {
 } from '../../app/StoreActionType'
 import type { DataContainer } from '../DataContainer'
 import loadCityContent from './loadCityContent'
+import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import DatabaseContext from '../DatabaseContext'
+
+function * getRootAvailableLanguages (
+  context: DatabaseContext,
+  loadCriterion: ContentLoadCriterion, dataContainer: DataContainer): Saga<Map<string, string>> {
+
+  if (loadCriterion.shouldLoadLanguages()) {
+    const languages = yield call(dataContainer.getLanguages, context)
+    return new Map<string, string>(languages
+      .map(language => [language.code, `/${context.cityCode}/${language.code}`]))
+  }
+
+  // If there are no loaded languages the result is an empty map because we do not have a root category
+  return new Map<string, string>()
+}
 
 function * fetchCategory (dataContainer: DataContainer, action: FetchCategoryActionType): Saga<void> {
-  const {city, language, path, depth, key, forceUpdate, shouldRefreshResources} = action.params
+  const { city, language, path, depth, key, criterion } = action.params
   try {
-    yield call(loadCityContent, dataContainer, city, language, forceUpdate, shouldRefreshResources)
+    const loadCriterion = new ContentLoadCriterion(criterion)
```

Comment:
Why are you creating a new `ContentLoadCriterion` here?

Reply:
There is a class and a plain js object version of it.